### PR TITLE
[OneDNN] Matmul kernel optimize for swin model

### PR DESCRIPTION
**Pull request by zhanglirong1999**
### PR types
Performance optimization

### PR changes
Others

### Description
OneDNN binary as post_ops, it does not support 1CHW to broadcast, which will casue regression issue. #59252  Thus manually use stride to fill it into NCHW.

### CODE REVIEW

**Comment by zhanglirong1999** [comment]
@xinyu-intel , @vivienfanghuagood , @yuanlehome , hi, can you help to review this PR?  Thanks~

File: paddle/phi/kernels/fusion/onednn/fused_matmul_kernel.cc [hunk 1]
```diff
@@ -193,6 +207,28 @@ class FusedMatmulOneDNNHandler
...
+    size_t size = this->fwd_pd_->dst_desc().get_size() / sizeof(XT);
+    XT *dst = static_cast<XT *>(src_mem->get_data_handle());
+
+#pragma omp parallel for
+    for (size_t i = 0; i < size; ++i) {
+      auto mod_i =
+          static_cast<int>(i - floor(i / (IC * IH * IW)) * (IC * IH * IW));
+      // Use stride to make 1*C*H*W to N*C*H*W to avoid broadcast overhead
+      dst[i] = input_data[mod_i];
+    }
```
Comment (xinyu-intel) [inline paddle/phi/kernels/fusion/onednn/fused_matmul_kernel.cc]: seems here changes the data handle instead of the stride?

Comment (LLee233) [reply paddle/phi/kernels/fusion/onednn/fused_matmul_kernel.cc]: Do you mean directly change the stride of mem? That is also a good method, but I am not sure if it's intuitive? This kind of change in PR can also attain a big improvement, though yours seems bring less overhead...

Comment (xinyu-intel) [reply paddle/phi/kernels/fusion/onednn/fused_matmul_kernel.cc]: I think both are okay, but the code change is not align with the comments. Maybe adjust the comments will be better. BTW, is openmp always available in Paddle?

Comment (LLee233) [reply paddle/phi/kernels/fusion/onednn/fused_matmul_kernel.cc]: Seems they diasble openmp in WIN32 case:
image
Maybe adding
`#if defined(_OPENMP)`
could be better.

**Review by yuanlehome — APPROVED** [review]
LGTM

**Review by xinyu-intel — APPROVED** [review]

**Pull request merged**